### PR TITLE
9 feat global defines

### LINKED ISSUE #9: Feat Global Defines

### PR DESCRIPTION
**Pull request by Philipp66904**

### CODE REVIEW

File: include/common_package/topic_names.hpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+namespace common_lib {
+
+namespace topic_names {
+
+// FCC Bridge
+constexpr const char* const BatteryState = "uav_battery_state";
+constexpr const char* const GPSPosition = "uav_gps_position";
```
Comment (J0las) [inline include/common_package/topic_names.hpp:10]:
```suggestion
constexpr const char* const GPSPosition = "uav_gps_position"; /**< Topic name for the GPS telemetry */
```

File: include/common_package/topic_names.hpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+constexpr const char* const MissionProgress = "uav_mission_progress";
+constexpr const char* const Pose = "uav_pose";
+constexpr const char* const RCState = "uav_rc_state";
+constexpr const char* const SafetyLimits = "uav_safety_limits";
+constexpr const char* const UAVCommand = "uav_command";
+constexpr const char* const UAVHealth = "uav_health";
+constexpr const char* const UAVWaypointCommand = "uav_waypoint_command";
+
```
Comment (J0las) [inline include/common_package/topic_names.hpp:19]:
```suggestion
constexpr const char *const FlightState = "uav_flight_state"; /**< Topic name for the UAV flight state telemetry */

```

File: include/common_package/topic_names.hpp [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+namespace common_lib {
+
+namespace topic_names {
+
+// FCC Bridge
+constexpr const char* const BatteryState = "uav_battery_state";
```
Comment (J0las) [inline include/common_package/topic_names.hpp:9]:
```suggestion
constexpr const char* const BatteryState = "uav_battery_state"; /**< Topic name for the UAV battery state telemetry */
```

File: include/common_package/topic_names.hpp [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+constexpr const char* const GPSPosition = "uav_gps_position";
+constexpr const char* const MissionProgress = "uav_mission_progress";
+constexpr const char* const Pose = "uav_pose";
+constexpr const char* const RCState = "uav_rc_state";
```
Comment (J0las) [inline include/common_package/topic_names.hpp:13]:
```suggestion
constexpr const char* const RCState = "uav_rc_state"; /**< Topic name for the RC state telemetry */
```

File: include/common_package/topic_names.hpp [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+namespace common_lib {
+
+namespace topic_names {
+
+// FCC Bridge
+constexpr const char* const BatteryState = "uav_battery_state";
+constexpr const char* const GPSPosition = "uav_gps_position";
+constexpr const char* const MissionProgress = "uav_mission_progress";
+constexpr const char* const Pose = "uav_pose";
```
Comment (J0las) [inline include/common_package/topic_names.hpp:12]:
```suggestion
constexpr const char* const Pose = "uav_pose"; /**< Topic name for the UAV pose euler angle telemetry */
```

File: include/common_package/topic_names.hpp [hunk 6]
```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+namespace common_lib {
+
+namespace topic_names {
+
+// FCC Bridge
+constexpr const char* const BatteryState = "uav_battery_state";
+constexpr const char* const GPSPosition = "uav_gps_position";
+constexpr const char* const MissionProgress = "uav_mission_progress";
```
Comment (J0las) [inline include/common_package/topic_names.hpp:11]:
```suggestion
constexpr const char* const MissionProgress = "uav_mission_progress"; /**< Topic name for the mission progress telemetry */
```

File: include/common_package/topic_names.hpp [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+constexpr const char* const Pose = "uav_pose";
+constexpr const char* const RCState = "uav_rc_state";
+constexpr const char* const SafetyLimits = "uav_safety_limits";
+constexpr const char* const UAVCommand = "uav_command";
+constexpr const char* const UAVHealth = "uav_health";
```
Comment (J0las) [inline include/common_package/topic_names.hpp:16]:
```suggestion
constexpr const char* const UAVHealth = "uav_health"; /**< Topic name for the uav health telemetry */
```

File: include/common_package/topic_names.hpp [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+constexpr const char* const UAVWaypointCommand = "uav_waypoint_command";
+
+// Mission Control
+constexpr const char* const Control = "control";
+constexpr const char* const Heartbeat = "heartbeat";
+constexpr const char* const JobFinished = "job_finished";
+constexpr const char* const MissionFinished = "mission_finished";
+constexpr const char* const MissionStart = "mission_start";
```
Comment (J0las) [inline include/common_package/topic_names.hpp:25]:
```suggestion
constexpr const char* const MissionStart = "mission_start"; /**< Topic name for the mission start signal send out when the FCC gets armed */
```

File: include/common_package/topic_names.hpp [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+constexpr const char* const Pose = "uav_pose";
+constexpr const char* const RCState = "uav_rc_state";
+constexpr const char* const SafetyLimits = "uav_safety_limits";
+constexpr const char* const UAVCommand = "uav_command";
+constexpr const char* const UAVHealth = "uav_health";
+constexpr const char* const UAVWaypointCommand = "uav_waypoint_command";
+
+// Mission Control
+constexpr const char* const Control = "control";
+constexpr const char* const Heartbeat = "heartbeat";
```
Comment (J0las) [inline include/common_package/topic_names.hpp:22]:
```suggestion
constexpr const char* const Heartbeat = "heartbeat"; /**< Topic name for the heartbeat */
```

File: include/common_package/topic_names.hpp [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+namespace topic_names {
+
+// FCC Bridge
+constexpr const char* const BatteryState = "uav_battery_state";
+constexpr const char* const GPSPosition = "uav_gps_position";
+constexpr const char* const MissionProgress = "uav_mission_progress";
+constexpr const char* const Pose = "uav_pose";
+constexpr const char* const RCState = "uav_rc_state";
+constexpr const char* const SafetyLimits = "uav_safety_limits";
+constexpr const char* const UAVCommand = "uav_command";
```
Comment (J0las) [inline include/common_package/topic_names.hpp:15]:
```suggestion
constexpr const char* const UAVCommand = "uav_command"; /**< Topic name for uav commands */
```

**Review by J0las — CHANGES_REQUESTED** [review]
The comments are optional, I just copied them out of my code base.
But I really need the flight state topic.

**Comment by Philipp66904** [comment]
I added the flight state topic. @J0las

**Comment by Philipp66904** [comment]
I think that it is pretty clear what these constants are, so I suggest to not add additional and kind of redundant comments.

File: include/common_package/topic_names.hpp (hunk 2 above)
Comment (Philipp66904) [reply include/common_package/topic_names.hpp:19]: added

**Review by finn-ehlers — APPROVED** [review]

**Review by J0las — APPROVED** [review]
Looks good.
Signed off by: Johan <job8197@thi.de>

**Pull request merged**